signer/src/storage/in_memory.rs

```diff
@@ -259,11 +260,66 @@ impl super::DbRead for SharedStore {
         let pending_deposit_requests = self
             .get_pending_deposit_requests(chain_tip, context_window)
             .await?;
-        let store = self.lock().await;
 
-        let threshold = threshold as usize;
+        let minimum_acceptable_reclaim_unlock_time = self
+            .get_bitcoin_block(chain_tip)
+            .await?
+            .map(|block| block.block_height as i32 + MINIMUM_RECLAIM_PROXIMITY_TO_CHAIN_TIP as i32)
+            .ok_or(Error::MissingBitcoinBlock(*chain_tip))?;
+
+        let chain_tip_ref = self
+            .get_bitcoin_block(chain_tip)
+            .await?
+            .map(|block| model::BitcoinBlockRef {
+                block_hash: *chain_tip,
+                block_height: block.block_height,
+            })
+            .unwrap();
+
+        // Get only the pending deposit requests that cannot be reclaimed too close to the
+        // current chain tip.
+        let mut acceptable_pending_deposit_requests: Vec<_> = Vec::new();
+        for deposit_request in pending_deposit_requests.iter() {
+            let mut maybe_canonical_bitcoin_block_with_transaction: Option<
+                model::BitcoinBlockHash,
+            > = None;
+            let bitcoin_blocks_with_transaction = self
+                .get_bitcoin_blocks_with_transaction(&deposit_request.txid)
+                .await?;
+            for block_hash in bitcoin_blocks_with_transaction {
+                let block_ref = self
+                    .get_bitcoin_block(&block_hash)
+                    .await?
+                    .map(|block| model::BitcoinBlockRef {
+                        block_hash,
+                        block_height: block.block_height,
+                    })
+                    .unwrap();
+                if self
+                    .in_canonical_bitcoin_blockchain(&chain_tip_ref, &block_ref)
+                    .await?
+                {
+                    maybe_canonical_bitcoin_block_with_transaction = Some(block_hash);
+                    break;
+                }
+            }
+            let canonical_bitcoin_block_with_transaction =
+                maybe_canonical_bitcoin_block_with_transaction.ok_or(Error::MissingBlock)?;
```

Comment:
since the unlock time is relative we need to know the height of the block it was contained in, but if this call fails we can't figure it out.

Reply:
In reality this can't happen because earlier parts of the function would almost certainly not function either, so this is meant to be a flag that something is wrong more than it is meant to be a error one catches. This is in the context of this being a test struct as well